pkg/lib/cockpit-components-modifications.jsx

```diff
@@ -37,11 +37,17 @@ const _ = cockpit.gettext;
  * Enables showing shell and ansible script. Shell one is mandatory and ansible one can be omitted.
  *
  */
-export const ModificationsExportDialog = ({ show, onClose, shell, ansible }) => {
+export const ModificationsExportDialog = ({ show, onClose, shell, ansible, name }) => {
     const [active_tab, setActiveTab] = React.useState("ansible");
     const [copied, setCopied] = React.useState(false);
     const [timeoutId, setTimeoutId] = React.useState(null);
 
+    React.useEffect(() => {
```

Comment:
This is a shared component, which shouldn't make policy decisions based on the OS name (also, `name` is too generic). Also, I'm really allergic to useEffect(), given how brittle it is. It's also not necessary here.

I suggest changing this component to simply not show an Ansible tab if `ansible` is false-y? (i.e. `null` or empty string), and putting that policy into kdump-view.jsx, i.e.
```jsx
ansible={ this.state.os_release.NAME?.includes("MicroOsS") ? null : exportAnsibleTask(...) }
```

Reply:
This works for me. Will change it

Reply:
Without rewriting a good portion of the component, I can't find a way to avoid using useEffect. Without useEffect using `setActiveTab` would trigger an infinite rerender since we are changing state inside the component.

Reply:
React.useState() is the initialization, that ought to run just once? I.e. `React.useState(ansible ? "ansible" : shell)` ?

Reply:
ah, that may work. I'll test it